### DISCUSSION TITLE
Craft 4 - Conditional Fields: Requires page refresh

### DISCUSSION THREAD

**Discussion by magnessjo** [solspace/craft-calendar#164] [Feature Requests]
**Describe the bug or issue you're experiencing**

Adding conditional fields to calendar fields doesn't work like the rest of the system.  I assume that Craft is doing a refresh for their revisions on change (field value changes).  The calendar plugin doesn't appear to have revisions built-in (maybe I am missing something?).

**Steps to reproduce**

See attached example.  For local testing - make a lightswitch field and set the entry condition for another field to the light switch.   

**Expected behavior**

Conditional fields should show without having to do a save

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 4.0.1 
- Calendar Version: 4.0.0-beta.3
- Calendar Edition: Pro
- Fresh Install or Upgrade: Upgrade 
- PHP - 8.1.3

**Additional context**

Apologizes in advance if I am missing something already built-in. 

https://user-images.githubusercontent.com/856595/167449562-a35c9487-007e-4b92-bfd6-7a8a84fdea75.mov

**Comment by kjmartens** [comment]
Thanks for reporting this, @magnessjo. We'll check into it soon. 🙂

**Comment by kjmartens** [comment]
Hi again @magnessjo,

We've spent a lot of time investigating this one, and it seems as though this might not be possible for third-party plugins yet, or it might require too much work to get it working right now.

**Comment by magnessjo** [comment]
Hi @kjmartens, Thanks for looking into it! My assumption is that it would be possible since it's an element type but I would love to know details if that's not the case (if possible).  

Do you have a public roadmap for the plugin?  I assume this will be part of the Craft 4 roadmap since it base functionality in Craft 4?  I'm looking for someway to track its progress outside of this ticket.  

I assume it will be a highly requested feature in Craft 4 so thanks again for looking into it!

**Comment by kjmartens** [comment]
@magnessjo My understanding is that this feature is tied into revision/drafts support, something else Calendar does not currently support. I believe that Revisions / Drafts support is still only available for native Element types like `Entry`, `Asset`, `User`, `Category`, etc. We tried simulating a lot of this manually but it doesn't fully work for every field type, etc. Perhaps there is a way to have this work, but there doesn't appear to be any documentation about this. We've also contacted the Craft team about it and should hear back soon. 

At any rate, this issue item will be updated if/when we are able to support Conditional rules, so you would be notified here if/when that changes. 🙂

**Comment by magnessjo** [comment]
Thank you @kjmartens!

**Comment by outline4** [comment]
Hi, I am using craft 4.5.11 and calendar 4.0.9.

This is still not working. 

Are you planning on adding this? 

IMHO Conditional fields are a huge plus! 

@magnessjo did you perhaps solve it somehow?

**Reply by kjmartens** [reply]
We hope to add support for this in the future, but it appears to require a lot of work to implement. Might be something we can address with Calendar 5 for Craft 5 though 🙂
